tests/with-docker-registry/Earthfile

```
@@ -37,23 +41,63 @@ docker-load-test:
     # Index is used to create parallel tests.
     ARG INDEX=0
     RUN echo "$INDEX"
+    WITH DOCKER --load +a-test-image
+        RUN docker run test-abc-img:xyz
+    END
+
+docker-load-shellout-test:
+    WITH DOCKER --load=+a-test-image-with-shell-out
+        RUN docker run test-img-with-shellout:myver | grep "you found me"
+    END
+
+docker-load-arg-test:
+    WITH DOCKER --load=(+a-test-image --name=foo --var bar)
+        RUN docker run test-foo-img:xyz | grep "hello bar"
+    END
+    WITH DOCKER --load=(+a-test-image \
+            --name=foo \
+            --var \
+            bar)
+        RUN docker run test-foo-img:xyz | grep "hello bar"
+    END
+    WITH DOCKER --load="(+a-test-image --name=bar --var buz)"
+        RUN docker run test-bar-img:xyz | grep "hello buz"
+    END
+    WITH DOCKER --load=other-name:latest=(+a-test-image --name=foo --var buz)
+        RUN docker run other-name:latest | grep "hello buz"
+    END
+    WITH DOCKER --load=other-name:latest=(+a-test-image \
+            --name=foo \
+            --var \
+            buz)
+        RUN docker run other-name:latest | grep "hello buz"
+    END
+    WITH DOCKER --load="other-name:latest=(+a-test-image --name=bar --var buz)"
+        RUN docker run other-name:latest | grep "hello buz"
+    END
+    WITH DOCKER --load=other-name:latest="(+a-test-image --name=bar --var buz)"
+        RUN docker run other-name:latest | grep "hello buz"
+    END
+
+docker-load-multi-test:
     WITH DOCKER \
-            --pull hello-world \
-            --load +a-test-image
-        RUN docker run test-abc-img:xyz && \
-            docker run hello-world
+        --load=(+another-test-image --INDEX=1) \
+        --load=(+another-test-image --INDEX=2) \
+        --load=(+another-test-image --INDEX=3) \
+        --load=(+another-test-image --INDEX=4) \
+        --load=(+another-test-image --INDEX=5)
+        RUN docker run --rm another-test-img:i1 && \
```

Is it possbile to add back in the grep which is used to ensure the `INDEX` arg gets correctly propigated into the container that's being built and run?
e.g.

```suggestion
        RUN docker run --rm another-test-img:i1 | grep "test img 1" && \
```

(and `grep 2`,...`grep 5`)